src/core/algorithms/dd/split/split.cpp

```diff
@@ -353,51 +404,54 @@ std::vector<DF> Split::SearchSpace(model::ColumnIndex index) {
+        }
+        return is_double;
+    };
 
     for (std::size_t row_index = 0; row_index < dif_num_rows; row_index++) {
         model::TypeId type_id = dif_column.GetValueTypeId(row_index);
         if (type_id == +model::TypeId::kString) {
-            std::string df_str;
-            if (dif_column.IsMixed()) {
-                model::MixedType const* mixed_type = dif_column.GetIfMixed();
-                std::byte const* df_string = dif_column.GetValue(row_index);
-                std::unique_ptr<model::Type> t = mixed_type->RetrieveType(df_string);
-                std::byte const* df_string_value = mixed_type->RetrieveValue(df_string);
-                model::StringType const* str_type = static_cast<model::StringType const*>(t.get());
-                df_str = str_type->ValueToString(df_string_value);
-            } else {
-                model::StringType const& str_type = static_cast<model::StringType const&>(type);
-                std::byte const* df_string = dif_column.GetValue(row_index);
-                df_str = str_type.ValueToString(df_string);
-            }
-
+            std::string df_str = dif_column.GetDataAsString(row_index);
             std::smatch matches;
             if (std::regex_match(df_str, matches, df_regex)) {
-                double const lower_limit = model::TypeConverter<double>::kConvert(matches[1].str());
-                double const upper_limit = model::TypeConverter<double>::kConvert(matches[3].str());
-
-                if (upper_limit >= min_max_dif_[index].lower_bound &&
-                    lower_limit <= min_max_dif_[index].upper_bound && lower_limit <= upper_limit) {
-                    model::DFConstraint intersect = {
-                            std::max(lower_limit, min_max_dif_[index].lower_bound),
-                            std::min(upper_limit, min_max_dif_[index].upper_bound)};
-                    if (intersect != min_max_dif_[index]) {
-                        limits.insert(intersect);
+                if (double_check(matches[1].str()) && double_check(matches[2].str())) {
+                    double const lower_limit =
+                            model::TypeConverter<double>::kConvert(matches[1].str());
+                    double const upper_limit =
+                            model::TypeConverter<double>::kConvert(matches[2].str());
+
+                    if (model::GreaterOrEqual(upper_limit, min_max_dif_[index].lower_bound) &&
+                        model::LessOrEqual(lower_limit, min_max_dif_[index].upper_bound) &&
+                        model::LessOrEqual(lower_limit, upper_limit)) {
+                        model::DFConstraint intersect = {
+                                std::max(lower_limit, min_max_dif_[index].lower_bound),
+                                std::min(upper_limit, min_max_dif_[index].upper_bound)};
+                        if (intersect != min_max_dif_[index]) {
+                            limits.insert(intersect);
+                        }
```

Comment:
same thing, we can create a method that will check intersection:
```cpp
    std::optional<DFConstraint> IntersectWith(DFConstraint const& other) const {
        double intersect_lower = std::max(lower_bound, other.lower_bound);
        double intersect_upper = std::min(upper_bound, other.upper_bound);

        if (LessOrEqual(intersect_lower, intersect_upper)) {
            return DFConstraint{intersect_lower, intersect_upper};
        }
        return std::nullopt;
    }
```

Reply:
Also to avoid a lot of `LessOrEqual` checks, just create a method `IsValid`:
```cpp
    bool IsValid() const {
        return LessOrEqual(lower_bound, upper_bound);
    }
```

Reply:
And we can use them like this:
```cpp
    double const lower_limit = //smth
    double const upper_limit = //smth
    model::DFConstraint parsed_limits{lower_limit, upper_limit};

    if (parsed_limits.IsValid()) {
        auto intersect = parsed_limits.IntersectWith(min_max_dif_[index]);
        if (intersect && *intersect != min_max_dif_[index]) {
            limits.insert(*intersect);
        }
    }
```